components/src/dynamo/frontend/sglang_prepost.py

```
@@ -116,6 +116,63 @@ def _normalize_sglang_parser_name(parser_name: str | None) -> str | None:
+    Kimi-K3 hit exactly that: ``reasoning_default='thinking'`` (thinking on
+    unless ``chat_template_kwargs.thinking is False``), but it was in neither
+    set and has no chat template, so reasoning was never enabled -- the parser
+    stayed off, ``reasoning_content`` came back null, and the raw
+    ``<|close|>think<|sep|>`` marker leaked into ``content``.
+
+    Ask SGLang instead. Returns None when the parser is unknown to this SGLang
+    build, in which case the caller falls back to the static tables.
+    """
+    try:
+        detector = ReasoningParser(model_type=parser_name).detector
+    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)